aten/src/ATen/native/cpu/MaxPoolKernel.cpp

```diff
@@ -88,6 +88,42 @@ void cpu_max_pool(
   }
 }
 
+template <typename scalar_t>
+vec::Vectorized<scalar_t> isnan(vec::Vectorized<scalar_t> vec) {
+  return vec.isnan();
+}
+
+template <>
+vec::Vectorized<unsigned char> isnan<unsigned char>(vec::Vectorized<unsigned char> vec) {
+  Vectorized<unsigned char> ret(false);
+  return ret;
+}
+
+template <>
+vec::Vectorized<signed char> isnan<signed char>(vec::Vectorized<signed char> vec) {
+  Vectorized<signed char> ret(false);
+  return ret;
+}
+
+template <>
+vec::Vectorized<short> isnan<short>(vec::Vectorized<short> vec) {
+  Vectorized<short> ret(false);
+  return ret;
+}
+
+template <>
+vec::Vectorized<int> isnan<int>(vec::Vectorized<int> vec) {
+  Vectorized<int> ret(false);
+  return ret;
+}
+
+template <>
+vec::Vectorized<int64_t> isnan<int64_t>(vec::Vectorized<int64_t> vec) {
+  Vectorized<int64_t> ret(false);
+  return ret;
+}
```

Comment:
Simplify these with SFINAE `std::is_integral` check?

Reply:
good point, thanks

Reply:
looks like it does not work somehow, I'll need to revert this